libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/route/line/api/MapboxRouteLineApi.kt

```diff
@@ -1440,7 +1442,7 @@ class MapboxRouteLineApi(
         routes.addAll(distinctNewRoutes)
         primaryRoute = distinctNewRoutes.firstOrNull()
         MapboxRouteLineUtils.trimRouteDataCacheToSize(size = distinctNewRoutes.size)
-        this.activeLegIndex = INVALID_ACTIVE_LEG_INDEX
+        this.activeLegIndex = activeLegIndex
```

Comment:
With this change you made me go and check whether it supports some workaround. I couldn't find any problems with it while testing, however, I found this:
1. On route progress updates we only update masking layers when leg changes;
2. Before your change the first route progress updates resulted in a leg change. Now it's not.
3. Here we have [this line of code](https://github.com/mapbox/mapbox-navigation-android/blob/e3192938438231825d3b255a35d137d06d2e3cd9/libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/route/line/api/MapboxRouteLineApi.kt#L1853), which probably has to use legIndex instead of defaulting to 0, so that in case someone sets routes with non-zero initial index, it works fine.